SurveyApp/Modules/SurveyDetail/SurveyDetailPresenter.swift

```diff
@@ -42,9 +47,14 @@ extension SurveyDetailPresenter: SurveyDetailViewOutput {
 
 extension SurveyDetailPresenter: SurveyDetailInteractorOutput {
     
-    // TODO: Will update with integration
-    func didPressStartSurvey() {
-        router.showSurveyQuestion()
+    func didFetchSurveyDetail(questions: [SurveyQuestion]) {
+        if let survey = survey {
+            router.showSurveyQuestion(id: survey.id, questions: questions)
+        }
```

Comment:
Following Mark's comment, this would be nice as a guard let too
```suggestion
        guard let survey = survey else { return }
        router.showSurveyQuestion(id: survey.id, questions: questions)
```